src/geventhttpclient/connectionpool.py

```diff
@@ -204,8 +204,13 @@ def _create_tcp_socket(self, family, socktype, protocol):
                 family, socktype, protocol)
 
             if self.ssl_context_factory is None:
-                ssl_options = self.default_options.copy()
-                ssl_options.update(self.ssl_options)
+                ssl_options = {}
+                if not self.insecure:
+                    ssl_options = self.default_options.copy()
+                    ssl_options.update(self.ssl_options)
```

Comment:
I think we have to define what "insecure" means. For me it just means no certificate checks, so just making sure that that CERT_REQUIRED is never part of the ssl options might already be insecure enough.

@ml31415 @gwik opinions?